hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -3086,10 +3090,66 @@ void syncBlock(RecoveringBlock rBlock, List<BlockRecord> syncList)
       datanodes[i] = r.id;
       storages[i] = r.storageID;
     }
-    nn.commitBlockSynchronization(block, newBlock.getGenerationStamp(),
+    commitBlockSyncWithRetry(nn, block, newBlock.getGenerationStamp(),
```

Comment:
This was explicitly needed for  EE where the datanode are slow to respond to the request as the block(s) would be downloaded/uploaded from/to the cloud. If we add this to CE then the code will be identical in both repos but this code will not be used in CE.

Reply:
@smkniazi Could that be the reason for receiving multiple commit messages to namenode, resulting in accounting for quota multiple times?